tests/clients/git/test_git.py

```diff
@@ -0,0 +1,175 @@
+import string
+
+from jumpscale.loader import j
+from tests.base_tests import BaseTests
+
+
+class GitTests(BaseTests):
+    def setUp(self):
+        super().setUp()
+        self.instance_name = self.random_name()
+        self.repo_dir = f"/tmp/{self.random_name()}"
+        self.repo_name = "js-ng"
+        self.repo_url = "https://github.com/threefoldtech/js-ng"
+        j.sals.fs.mkdir(f"{self.repo_dir}")
+        j.sals.process.execute(f"git clone {self.repo_url}", cwd=f"{self.repo_dir}")
+        self.git_client = j.clients.git.new(name=self.instance_name, path=f"{self.repo_dir}/{self.repo_name}/")
+
+    def tearDown(self):
+        j.clients.git.delete(self.instance_name)
+        j.sals.fs.rmtree(path=f"{self.repo_dir}")
+
+    def random_name(self):
+        return j.data.idgenerator.nfromchoices(10, string.ascii_letters)
+
+    def test01_check_git_config(self):
+        """Test case for checking git config file.
+
+        **Test Scenario**
+        - Get a git client.
+        - Read the git.config file.
+        - Check that remote_url equal to repo_url.
+        - Check that url equal to remote_url.
+        """
+        self.info("Read the git.config file")
+        git_config = j.sals.fs.read_file(f"{self.repo_dir}/{self.repo_name}/.git/config")
+
+        self.info("Check that remote_url equal to repo_url")
+        self.assertEqual(self.repo_url, self.git_client.remote_url)
+
+        self.info("Check that git config url equal to remote_url")
+        self.assertIn(self.git_client.remote_url, git_config)
+
+    def test02_git_branch(self):
+        """Test case for checking a branch name.
+
+        **Test Scenario**
+        - Get a git client.
+        - Get the branch name.
+        - Check branch name.
+        """
+        self.info("Get the branch name")
+        branch_name = j.sals.process.execute("git branch", cwd=f"{self.repo_dir}/{self.repo_name}")
+
+        self.info("Check branch name")
+        self.assertIn(self.git_client.branch_name, branch_name[1])
```

Comment:
`git branch` return all branch and maybe the current branch is not the one returned from git client.

Reply:
`git branch` put * before the current branch.

Reply:
can use git branch --show-current ?

Reply:
Sure